src/System.Linq.Expressions/tests/TypeBinary/TypeBinaryTests.cs

```diff
@@ -50,6 +50,12 @@ private static IEnumerable<Expression> Constants
                 yield return Expression.New(typeof(string).GetConstructor(new[] { typeof(char[]) }), Expression.Constant(new[] { 't', 'e', 's', 't' }));
                 yield return Expression.Convert(Expression.Constant(DateTime.MaxValue), typeof(DateTime?));
                 yield return Expression.Constant(1, typeof(int?));
+                yield return Expression.Constant(1, typeof(object));
+                yield return Expression.Constant(1, typeof(IComparable));
+                yield return Expression.Constant(DateTime.MinValue, typeof(object));
+                yield return Expression.Constant(1, typeof(ValueType));
+                yield return Expression.Constant(DateTime.MinValue, typeof(ValueType));
+                yield return Expression.Constant(DateTime.MinValue, typeof(IComparable));
```

Comment:
Probably makes sense to add to the list of types some common 32/8bit enums and nullables for those enums. Enums often cause troubles in casts.

Also, for completeness, a constant of derived reference type typed as base reference type - like NullRef instance types as System.Exception.